doc: 添加贡献指南 (#386)
MCDReforged/PluginCatalogue: CONTRIBUTING.md

**English** | [中文](CONTRIBUTING_cn.md)

# Contribution Guidelines

We are truly honored and grateful for your willingness to contribute to this project.

To ensure that your efforts integrate smoothly into the project and benefit the entire community, we have created this guide to help you better understand our standards and processes.

> We recommend you to carefully reading the following information, which is designed to help you work efficiently and ensure that your contributions meet our requirements and guidelines.

Reviewer: ```suggestion
We recommend you carefully read the following information, which is designed to help you work efficiently and ensure that your contributions meet our requirements and guidelines.
```

Sections marked with an `*` are **requirements**, while the rest are **suggestions**, both serving as references for your development process and for catalogue maintainers when reviewing PRs.

We hope this guide can be a great starting point for you!

## Adding or modifying plugin?

> ### Is your plugin suitable?

Reviewer: ```suggestion
### Is your plugin suitable for the catalogue?
```

> Do not submit plugins in a hurry.
> 
> Before creating and submitting new plugins, consider following questions:

Reviewer: ```suggestion
Do not submit plugins in a hurry. Before creating and submitting new plugins, consider the following questions:
```

- Is there already a plugin with really similar functionality? —— Unnecessarily reinventing wheels is not desirable;

> - Is your code necessary to provide as a MCDR plugin? Would it be more suitable for PyPI? —— Consider the use purpose and make sure it is necessary;

Reviewer: ```suggestion
- Is your code necessary to provide as an MCDR plugin? Would it be more suitable for PyPI? —— Consider the use purpose and make sure it is necessary;
```

> - Is the plugin general enough? Can new users have reasons to use your plugin? —— Do not submit self-use only or completely useless plugins;

Reviewer: ```suggestion
- Is the plugin commonly applicable? Can new users have reasons to use your plugin? —— Do not submit self-use only or completely useless plugins;
```
```suggestion
- Is the plugin generally used? Can new users have reasons to use your plugin? —— Do not submit self-use only or completely useless plugins;
```

> - Is your project ready for public testing (beta)? —— If it is still in an early version, wait until it is stable.

Reviewer: ```suggestion
- Is your project ready for public testing (beta)? —— If it is still in an early stage, wait until it ready to be published.
```

### *Make sure you have permissions

> Users submitting plugins to the plugin catalogue should be in one of the following capacities:
> 
> - Author, maintainer, or collaborator of the plugin;

Reviewer: ```suggestion
Users submitting plugins to the plugin catalogue should be in one of the following capacities:
- Author, maintainer, or collaborator of the plugin;
```

> - An uploader who has explicit permission from someone above.

Reviewer: ```suggestion
- An uploader who has explicit permission from someone above in writing.
```

> If your plugin reference or use codes from other projects:
> - If the codes is open source, follow their open source licence;
> - If the codes is proprietary, make sure you have the right to use them, and follow their agreement.

Reviewer: ```suggestion
If your plugin references or uses codes from other projects:
- If the codes is open source, follow their open source license;
- If the codes is proprietary, make sure you have the right to use them, and follow their agreement.
```

NO British English :(

### *Uniqueness and compatibility

- Plugin name and ID should not be **too similar** to other plugins;
- Explicitly declare required plugins and/or Python packages in corresponding fields (also declare MCDR version if needed);
- Make sure that your plugin works properly on target platforms.

### *Write plugin information correctly

- Create or edit `plugin_info.json` as described in [documentation](https://mcdreforged.readthedocs.io/en/latest/plugin_dev/plugin_catalogue.html);
- Take care to check the `label` field is appropriate by referring to existing plugins;

> - Plugin IDs must be consistent in all fields;

Reviewer: ```suggestion
- Plugin ID must be consistent at all places;
```

- `description` and `introduction` fields should provide at least in `en_us`;
- The plugin README should be placed in `related_path` and named `README.md` (case insensitive), bilingual version is recommended but not required.

### Introduce your plugin properly

A good and detailed introduction is one of the conditions for becoming an excellent plugin.

- In the README or documentation, explain the plugin's functions, features, usage, etc. in detail;
- In the `description`, briefly summarise what the plugin does in one sentence;

> - Introduce the plugin's features in the `introduction`, or direct reference it to the README file.

Reviewer: 建议设置为强制要求

> 
> 

Reviewer: ```suggestion

```

Reviewer: 方便未来修改比对，确保每行一致
